DashAI/front/src/api/datasets.ts

```diff
@@ -13,7 +13,18 @@ export const getDatasets = async (): Promise<IDataset[]> => {
   return response.data;
 };
 
+export const updateDataset = async (
+  id: number,
+  name: string,
+  taskName: string,
+): Promise<IDataset> => {
+  const response = await api.patch<IDataset>(
+    `${datasetEndpoint}/${id}?name=${name}&task_name=${taskName}`,
+  );
+  return response.data;
```

Comment:
As convention, POST, PUT, PATCH should deliver the new data in the request body.
On the other hand, when the resource is already created and you want to modify it (in PUT and PATCH), it must be accessed through Paths (i.e., datasets/1, etc.).

To get an idea, this could be read as: 
_Update the dataset with id=1 with the json data provided in the body._

```suggestion
  const response = await api.patch<IDataset>(
    `${datasetEndpoint}/${id}`,
    {"name": name, "task_name":  task_name}
  );
  return response.data;
```

Reply:
Thank you!

I fixed it using the same method as for solving the post request in the case of uploading an experiment, passing the elements of the object as formData.

The modifications on the backend to make it work are similar: add `Form(...)` to each parameter that requires it.